### DISCUSSION TITLE
Bonsai.ML 0.1.0

### DISCUSSION THREAD

**Discussion by glopesdev** [bonsai-rx/bonsai#1709] [Announcements]
We are excited to announce [Bonsai.ML](https://bonsai-rx.org/machinelearning), a new project to empower Bonsai workflows with state-of-the-art machine learning functionality.

In this initial release, we introduce the [Bonsai.ML.LinearDynamicalSystems](https://www.nuget.org/packages/Bonsai.ML.LinearDynamicalSystems) package containing one of the most used statistical models for time series, the linear dynamical systems model (LDS). The LDS model can be used to infer continuous latent variables from time series observations, and we provide a first application of its use for estimating kinematic variables of tracked objects.

In the kinematics model, the time series observations are noisy measurements of an object position from which the LDS model can then infer denoised position, velocity and acceleration vectors of the object motion. Three examples of its use with both experimental and simulated data are available in the [Examples Page](https://bonsai-rx.org/machinelearning/examples/README.html) to help you quickly get started applying the model to your own tracking data.

We hope that the Bonsai.ML project will attract contributions from machine learning researchers and developers who might be interested in making their methods available for online inference in Bonsai. Because many machine learning method developers use languages other than C#, like Python, R or Julia, we are building packages to facilitate the interaction between Bonsai and these other languages.

Earlier we introduced the new [Python Scripting](https://bonsai-rx.org/python-scripting) package that allows full access to modern Python libraries up to and including Python 3.12. [Bonsai.ML.LinearDynamicalSystems](https://bonsai-rx.org/machinelearning/articles/LinearDynamicalSystems/lds-overview.html) also demonstrates the use of the Python scripting package, as it directly imports the linear dynamical systems functionality in the [lds_python](https://github.com/joacorapela/lds_python) package.

We will be adding more machine learning models (e.g., hidden Markov models), examples, and other language interface packages to Bonsai.ML. If you experience any difficulty in the use of these packages feel free to open an issue on the [Bonsai.ML GitHub repository](https://bonsai-rx.org/machinelearning) or start a discussion here. We are looking forward to hearing from you any feedback about Bonsai.ML.

The Bonsai.ML team:
- @ncguilbeault 
- @glopesdev 
- @joacorapela

**Comment by sneakers-the-rat** [comment]
Congrats @glopesdev and team!!! Looks really good!!!!!!!
